internal/cephfs/store/backingsnapshot.go

```diff
@@ -170,3 +170,70 @@ func UnrefSelfInSnapshotBackedVolumes(
 		},
 	)
 }
+
+func UpdateRefCount(
+	ctx context.Context,
+	volOptions *VolumeOptions,
+	clusterName string,
+	setMetadata bool,
+	secrets map[string]string,
+) error {
+	ioctx, err := volOptions.conn.GetIoctx(volOptions.MetadataPool)
+	if err != nil {
+		log.ErrorLog(ctx, "failed to create RADOS ioctx: %s", err)
+
+		return err
+	}
+	defer ioctx.Destroy()
+
+	ioctx.SetNamespace(fsutil.RadosNamespace)
+
+	var (
+		backingSnapID = volOptions.BackingSnapshotID
+		ioctxW        = radoswrapper.NewIOContext(ioctx)
+	)
+
+	created, err := reftracker.Add(
+		ioctxW,
+		fmtBackingSnapshotReftrackerName(backingSnapID),
+		map[string]struct{}{
+			backingSnapID:    {},
+			volOptions.VolID: {},
+		},
+	)
+	if err != nil {
+		log.ErrorLog(ctx, "failed to add refs for backing snapshot %s: %v",
+			backingSnapID, err)
+
+		return err
+	}
+
+	defer func() {
+		if err == nil {
+			return
+		}
+
+		// Clean up after failure.
+
+		var deleted bool
+		deleted, err = reftracker.Remove(
+			ioctxW,
+			fmtBackingSnapshotReftrackerName(backingSnapID),
+			map[string]reftype.RefType{
+				backingSnapID:    reftype.Normal,
+				volOptions.VolID: reftype.Normal,
+			},
+		)
+		if err != nil {
+			log.ErrorLog(ctx, "failed to remove refs in cleanup procedure for backing snapshot %s: %v",
+				backingSnapID, err)
+		}
+
+		if created && !deleted {
+			log.ErrorLog(ctx, "orphaned reftracker object %s (pool %s, namespace %s)",
+				backingSnapID, volOptions.MetadataPool, fsutil.RadosNamespace)
+		}
+	}()
```

Comment:
i dont think this will get executed in error case as you are already returning at line 208

Reply:
Yes right, updated. Hope that will help.